shoop/front/middleware.py

```diff
@@ -91,15 +87,22 @@ def _set_timezone(self, request):
             timezone.activate(request.person.timezone)
             # TODO: Fallback to request.shop.timezone (and add such field)
 
-    def _get_maintenance_response(self, request):
-        if request.shop.maintenance_mode and not request.user.is_superuser:
-            return HttpResponse(loader.render_to_string("shoop/front/maintenance.jinja", request=request), status=503)
-
     def process_response(self, request, response):
         if hasattr(request, "basket") and request.basket.dirty:
             request.basket.save()
+        maintenance_response = self._get_maintenance_response(request)
```

Comment:
Now we're back to the actual view doing its work before we override the response with another. :\